src/components/modals/ModalComponent.tsx

```diff
@@ -66,9 +68,12 @@ class ModalComponent extends React.Component<ModalProps> {
     window.addEventListener('keydown', this.onKeyDown, true)
   }
 
-  close = () => {
+  close = (fullReload: any = false) => {
     this.animateAndClose!()
     this.props.onClose?.()
+    if (Capacitor.isNativePlatform() && fullReload) {
+      window.location.reload()
```

Comment:
I don't believe we should be doing a full page reload here. The only cases where we have those at the moment are for unrecoverable error handling.

Is there a particular reason for doing this? If I understand correctly, it's doing a full reload every time the Settings modal is closed.

Reply:
We have already had discussions about this in this pr. Its unfortunate that we have to do this, but it is necessary. We can only limit the changes down to the very minimal as possible. (I.E. Only in the settings menu with a flag set as true.)

Reply:
This is due to a limitation in changing the background color through the WebView. It's bad, but it's the only way to update the overscroll background color from what I understand.

It should reload the page only when the theme setting has changed.